 File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\werkzeug\middleware\proxy_fix.py", line 183, in __call__
    return self.app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\CyberSecurity\CyberTrack-main\routes.py", line 152, in analytics
    return render_template('analytics.html',
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "e:\CyberSecurity\CyberTrack-main\templates\analytics.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "e:\CyberSecurity\CyberTrack-main\templates\base.html", line 72, in top-level template code
    {% block content %}{% endblock %}
  File "e:\CyberSecurity\CyberTrack-main\templates\analytics.html", line 134, in block 'content'
    style="width: {{ "%.1f"|format(progress) }}%">
  File "C:\Users\moame\AppData\Roaming\Python\Python312\site-packages\jinja2\filters.py", line 1043, in do_format
    return soft_str(value) % (kwargs or args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'progress' is undefined